server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamOptions.java

```diff
@@ -93,4 +101,32 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
     public static DataStreamOptions fromXContent(XContentParser parser) throws IOException {
         return PARSER.parse(parser, null);
     }
+
+    /**
+     * Creates a class that composes the different fields of the data stream options and normalises explicitly nullified fields.
```

Comment:
Can you expand this javadoc and the one below it with more details about how it composes these? I think it'd be helpful for devs reading the code